pkg/expression/errors.go

```diff
@@ -76,12 +76,8 @@ func handleInvalidTimeError(ctx EvalContext, err error) error {
 		types.ErrDatetimeFunctionOverflow.Equal(err) || types.ErrIncorrectDatetimeValue.Equal(err)) {
 		return err
 	}
-	sc := ctx.GetSessionVars().StmtCtx
-	err = sc.HandleTruncate(err)
-	if ctx.GetSessionVars().StrictSQLMode && (sc.InInsertStmt || sc.InUpdateStmt || sc.InDeleteStmt) {
-		return err
-	}
-	return nil
+	tc := ctx.GetSessionVars().StmtCtx.TypeCtx()
+	return tc.HandleTruncate(err)
```

Comment:
Why do we use typectx, but not errctx here?🤔

Reply:
Because the old code is using `HandleTruncate`, using `errctx` is still ok